### PR TITLE
MSP-5178

### PR DESCRIPTION
**Pull request by marinhekman**

### CODE REVIEW

File: api/v1/Plan.php [hunk 1]
```diff
@@ -1158,16 +1158,16 @@ private function GetBaseGeometryInformation(
         $baseInfo["geometry_id"] = $geometryId;
 
         $baseInfoQuery = $this->getDatabase()->query(
-            "SELECT geometry_id, geometry_persistent, geometry_mspid FROM geometry WHERE geometry_id = ?",
+            "SELECT geometry_id, geometry_persistent FROM geometry WHERE geometry_id = ?",
```
Comment (marinhekman) [inline api/v1/Plan.php:1161]: geometry_mspid was unused here

File: api/v1/Plan.php [hunk 2]
```diff
@@ -1158,16 +1158,16 @@ private function GetBaseGeometryInformation(
...
             array($geometryId)
         );
         $persistentId = $baseInfoQuery[0]["geometry_persistent"];
         $mspIdQuery = $this->getDatabase()->query(
-            "SELECT geometry_mspid FROM geometry WHERE geometry_persistent = ? AND geometry_mspid IS NOT NULL",
+            "SELECT IFNULL(geometry_mspid,0) as geometry_mspid FROM geometry WHERE geometry_persistent = ?",
```
Comment (marinhekman) [inline api/v1/Plan.php:1166]: always return a geometry_mspid, even if it NULL in the database. Then it returns 0

File: api/v1/Plan.php [hunk 3]
```diff
@@ -1158,16 +1158,16 @@ private function GetBaseGeometryInformation(
...
             array($persistentId)
         );
         if (count($mspIdQuery) > 0) {
-            $baseInfo["geometry_mspid"] = $mspIdQuery[0]["geometry_mspid"];
+            $baseInfo["geometry_mspid"] = (int)$mspIdQuery[0]["geometry_mspid"];
```
Comment (marinhekman) [inline api/v1/Plan.php:1170]: force int for mspid

File: api/v1/Plan.php [hunk 4]
```diff
@@ -1279,7 +1279,7 @@ public function Export(array &$result, ?array &$errors = null): bool
 
                 $l['deleted'] = $this->getDatabase()->query(
                     "
-                    SELECT geometry_id
+                    SELECT geometry_id, 0 as geometry_mspid
```
Comment (marinhekman) [inline api/v1/Plan.php:1282]: just return 0 for the mspid for "deleted"
This seems to be the default for all the sea basins with starting plans

File: api/v1/Plan.php [hunk 5]
```diff
@@ -1340,6 +1340,10 @@ public function Export(array &$result, ?array &$errors = null): bool
 						",
                         array($geom['geometry_id'])
                     );
+                    $energyOutput = array_map(function($el) {
```
Comment (marinhekman) [inline api/v1/Plan.php:1343]: force int for maxcapacity

File: api/v1/Plan.php [hunk 6]
```diff
@@ -1360,7 +1364,10 @@ public function Export(array &$result, ?array &$errors = null): bool
                     ",
                     array($grid['grid_id'])
                 );
-
+                $grid['energy'] = array_map(function($el) {
```
Comment (marinhekman) [inline api/v1/Plan.php:1367]: force int for expected

File: src/Domain/SessionConfigJSONSchema.json [hunk 7]
```diff
@@ -98,7 +98,14 @@
                 "type": "number"
               },
               "plan_type": {
-                "type": "string"
+                "oneOf": [
```
Comment (marinhekman) [inline src/Domain/SessionConfigJSONSchema.json:101]: we accept 2 formats for plan_type. The old style still which is x,x,x. e.g. 0,0,0 or 0,1,0 etc
Or just an int which is then processed as bitwise flags

**Review by HutchinsonKevin — APPROVED** [review]

**Pull request merged**